provider/core.rb

```diff
@@ -63,7 +63,8 @@ def copy_files(output_folder)
       copy_file_list(output_folder, @config.files.copy)
     end
 
-    # version_name is used within templates because binding is passed down
+    # version_name is actually used because all of the variables in scope in this method
```

Comment:
Why a comment/rubocop disable instead of the underscore prefix?

@rambleraptor for ruby style knowledge also in case you have an opinion because I'm not sure which is actually better

Reply:
The Ruby convention is that anything with an underscore at the beginning is unused.

Example: a function with multiple parameters and the function code only uses the first parameter. All other parameters will begin with an _ (or just be an underscore)

Reply:
The problem is that in [parse in config.rb](https://github.com/GoogleCloudPlatform/magic-modules/pull/942/files#diff-ed8ed439ac92b9d32368d0c46f1b946dR79) rubocop will complain if it's prefixed with a version name because it's used in an unless statement to guard against nil. But here there isn't a reference so it will complain if there isn't a prefix.

However the crux is that all of the variables that are in local scope of the method are passed into our templating logic as 'local variables' as well. So all the variables need to be named consistently. So in this method we are calling `compile` which is the method that converts all 'local' variables into template accessible variables.

Reply:
Got it. Can you update the comment to say why the underscore prefix doesn't work? (the fact that rubocop will complain elsewhere)